src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -5782,11 +5782,13 @@ private static void CompleteFormatViewByInferredType(CompletionContext context,
                                 && uniqueNames.Add(viewDefinition.name)
                                 && viewPattern.IsMatch(viewDefinition.name))
                             {
-                                string completionText = quote != string.Empty
-                                    ? quote + viewDefinition.name + quote
-                                    : viewDefinition.name.IndexOfAny(s_charactersRequiringQuotes) != -1
-                                        ? "'" + viewDefinition.name + "'"
-                                        : viewDefinition.name;
+                                string completionText = viewDefinition.name;
+                                var quoteInUse = quote == string.Empty ? "'" : quote;
+                                if (quoteInUse == "'")
+                                {
+                                    completionText = completionText.Replace("'", "''");
+                                }
+                                completionText = quoteInUse + completionText + quoteInUse;
```

Comment:
This is the same as this right?

```suggestion
                                // If we need quotes, convert to single quotes and escape any existing ones
                                string completionText = quote == string.Empty
                                    ? viewDefinition.name
                                    : "'" + viewDefinition.name.Replace("'", "''") + "'";
```

Reply:
Or if we need to keep the check for characters in need of escaping:

```suggestion
                                 // If we need quotes, convert to single quotes and escape any existing ones
                                string completionText = (quote == string.Empty && viewDefinition.name.IndexOfAny(s_charactersRequiringQuotes) < 0)
                                    ? viewDefinition.name
                                    : "'" + viewDefinition.name.Replace("'", "''") + "'";
```

Reply:
I think the logic here still needs reworking

Reply:
I use the pattern which we use in other places. I think it makes code more understandable.